### PR TITLE
Clean up event naming

### PR DESCRIPTION
**Pull request by markus-wa**
See #45

Cc @micvbang - Any feedback appreciated.

I also decided to switch from past to present tense, removing many 'ed' suffixes, except where necessary so the events aren't confused with something else.

E.g. if it were `BombDefuse` instead of `BombDefused` it could be mistaken for the start of defusing.

### CODE REVIEW

**Comment by micvbang** [comment]
Looks good!

I'm not entirely sure about the change to present tense. Hmm. Maybe I've just gotten too used to the past tense, but I feel like it sounds better.

Is it on purpose that `GrenadeEvent` wasn't renamed to `Grenade`? I think that would make sense :slightly_smiling_face: Along with `GrenadeEventIf` -> `GrenadeIf`

**Comment by markus-wa** [comment]
I had two main reasons for favoring present tense.
1. At the point where the event is fired, it's 'happening' in that instant, not in the past (possibly even just *starting* to happen).
    1. The event may arguably be in the future, if we're in the middle of the demo.
2. The demo's internal events are present-tense as well

Examples:

1.
`SmokeExpire`: "The smoke is expiring now"
I assume that this signals the game to start fading away the smoke

`WeaponFire`: "A weapon is BEING fired" (even tho it has 'ed' here it is still present tense.
This would probably instruct the game to *start* rendering the muzzle flash and playing weapon sounds.

2.
```
GameEvent: smokegrenade_expired Data: map[userid:248 entityid:347 x:948.243 y:266.19666 z:1616.0312]
GameEvent: weapon_fire Data: map[weapon:weapon_m4a1 silenced:false userid:249]
GameEvent: player_footstep Data: map[userid:254]
GameEvent: weapon_fire Data: map[silenced:false userid:233 weapon:weapon_ak47]
GameEvent: weapon_fire Data: map[silenced:false userid:249 weapon:weapon_m4a1]
GameEvent: weapon_fire Data: map[weapon:weapon_ak47 silenced:false userid:233]
GameEvent: player_footstep Data: map[userid:236]
GameEvent: player_footstep Data: map[userid:254]
```

On the `GrenadeEvent`, that's on purpose, so it wouldn't be confused with being an actual grenade for some reason. I'd prefer to keep it like this. Same with the `BombEvent`

**Comment by micvbang** [comment]
Yeah, I appreciate those points.

Haha, that's funny. I always assumed the opposite of you, that grenades were gone by the time the expired event was thrown.

I just did a very scientific experiment :microscope:, staring at a replay of a smoke for a while. From what I can tell, the smoke actually begins to fade _before_ the SmokeExpire event is thrown, and is completely gone by the time it's thrown.

Maybe it makes sense to differentiate tense used for each event, based on what actually happens in-game?

Btw., the assumption about gunfire seems to be accurate!

**Comment by markus-wa** [comment]
Very interesting, thanks for analyzing that!

In that case I agree, the tense should fit to what is correct for each case.

**Comment by markus-wa** [comment]
I changed some more stuff, let me know if you see anything else
Otherwise I think I'll merge it this WE.

File: events/events.go [hunk 1]
```diff
@@ -1,4 +1,8 @@
 // Package events contains all events that can be sent out from demoinfocs.Parser.
+//
+// Events are generally named in the tense that fits the best for each event.
+// E.g. SmokeExpired is in the past tense because it's sent out when the smoke has completely faded away
+// while SmokeStart is in the present tense because it's sent out when the smoke starts to bloom.
```
Comment (micvbang) [inline events/events.go]: This is excellent!

**Comment by micvbang** [comment]
Sorry, I completely forgot about this PR. Once again great work, looking forward to using it!

**Comment by markus-wa** [comment]
No worries 😄 and thanks!

**Pull request merged**